From 2d4e61dab0e2461d6117713415a91ea922b2bd34 Mon Sep 17 00:00:00 2001
From: Marinus Schraal <mschraal@gnome.org>
Date: Wed, 21 Oct 2020 16:00:54 +0200
Subject: [PATCH] grltrackerwrapper: Fix artist queries to retrieve art

Artwork for artists was not retrieved anymore, this was due to the
artist field of the Grl.Media no longer be filled. This can be traced
back to bf973c1255, the initial porting of the Tracker artist queries to
Tracker 3.

In Music a distinction is not made between artist and albumartist tags.
The latter is always preferred, but might not be available. So, the
artist tag should always be considered filled. In this case however the
query was altered to only provide albumartist to the Grl.Media.

The art lookup resolve operations require the artist field of the
Grl.Media to be present, so they would return empty as the Grilo plugins
do not support albumartist.

Restore the previous behaviour of the queries.
---
 gnomemusic/grilowrappers/grltrackerwrapper.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gnomemusic/grilowrappers/grltrackerwrapper.py b/gnomemusic/grilowrappers/grltrackerwrapper.py
index 96b19897d..87189e3b9 100644
--- a/gnomemusic/grilowrappers/grltrackerwrapper.py
+++ b/gnomemusic/grilowrappers/grltrackerwrapper.py
@@ -276,7 +276,7 @@ class GrlTrackerWrapper(GObject.GObject):
                     SELECT
                         %(media_type)s AS ?type
                         (COALESCE(?album_artist, ?artist) AS ?id)
-                        ?artist_bind AS ?albumArtist
+                        ?artist_bind AS ?artist
                     WHERE {
                         ?song a nmm:MusicPiece;
                                 nmm:musicAlbum ?album;
@@ -599,14 +599,14 @@ class GrlTrackerWrapper(GObject.GObject):
                 artists_added.clear()
 
         query = """
-        SELECT ?type ?id ?albumArtist
+        SELECT ?type ?id ?artist
         WHERE {
             SERVICE <dbus:%(miner_fs_busname)s> {
                 GRAPH tracker:Audio {
                     SELECT
                        %(media_type)s AS ?type
                        (COALESCE(?album_artist, ?artist) AS ?id)
-                       ?artist_bind AS ?albumArtist
+                       ?artist_bind AS ?artist
                     WHERE {
                         ?song a nmm:MusicPiece;
                                 nmm:musicAlbum ?album;
-- 
GitLab

